src/commands/Moderation/softban.js

```diff
@@ -11,22 +11,46 @@ module.exports = class extends Command {
             requiredPermissions: ["USE_EXTERNAL_EMOJIS", "BAN_MEMBERS"],
             description: language => language.get("COMMAND_SOFTBAN_DESCRIPTION"),
             quotedStringSupport: false,
-            usage: "<member:membername> [days:int{1,7}] [reason:string] [...]",
+            usage: "<member:membername> [reason:string] [...]",
             usageDelim: " ",
-            extendedHelp: "No extended help available."
+            extendedHelp: "--messages <1-7> - Days of messages"
         });
     }
 
-    async run(msg, [member, days = 1, ...reason]) {
-        if (member.id === msg.author.id) return msg.reply(`${this.client.emotes.cross} ***${msg.language.get("MESSAGE_BAN_YOURSELF")}***`);
-        if (member.id === this.client.user.id) return msg.reply(`${this.client.emotes.cross} ***${msg.language.get("MESSAGE_BAN_PENGU")}***`);
-        if (!member.bannable) return msg.reply(`${this.client.emotes.cross} ***${msg.language.get("MESSAGE_BAN_CANT")}***`);
+    // eslint-disable-next-line complexity
+    async run(msg, [user, ...reason]) {
+        reason = reason ? reason.join(" ") : null;
 
-        reason = reason.length > 0 ? `${reason.join(" ")}\nBanned By: ${msg.author.tag}` : `No reason specified.\nBanned By: ${msg.author.tag}`;
-        await member.ban({ days: days, reason: reason });
-        msg.guild.members.unban(member.user.id);
+        if (user.id === msg.author.id) return msg.reply(`${this.client.emotes.cross} ***${msg.language.get("MESSAGE_BAN_YOURSELF")}***`);
+        if (user.id === this.client.user.id) return msg.reply(`${this.client.emotes.cross} ***${msg.language.get("MESSAGE_BAN_PENGU")}***`);
 
-        return msg.sendMessage(`${this.client.emotes.check} ***${member.user.tag} ${msg.language.get("MESSAGE_SOFTBANNED")}***`);
+        let target = await msg.guild.members.fetch(user.id).catch(() => null);
+
+        if (target) {
+            if (target.roles.highest.position >= msg.member.roles.highest.position) {
+                return msg.reply(`${this.client.emotes.cross} ***Target member is higher in role hierarchy than you.***`);
+            } else if (!target.bannable) {
+                return msg.reply(`${this.client.emotes.cross} ***${msg.language.get("MESSAGE_BAN_CANT")}***`);
+            }
+        } else {
+            target = user;
+        }
+
+        const msgDays = msg.flags.messages || msg.flags.msg || 0;
```

Comment:
Have just one flag instead of complicated it.
should be 

```javascript
const msgDays = "messages" in msg.flags ? Number(msg.flags.messages) : 0;
```

Reply:
Says should be 7 and not configurable. the point of softban is to delete all messages

Reply:
A flag wouldn't hurt, not that many people would use it anyways.

Reply:
not what im saying?